pkg/ipamd/datastore/data_store.go

```diff
@@ -145,38 +148,56 @@ type ENI struct {
 	IsTrunk bool
 	// IsEFA indicates whether this ENI is tagged as an EFA
 	IsEFA bool
-	// IsPDEnabled indicates whether prefix delegation is enabled
-	IsPDEnabled bool
 	// DeviceNumber is the device number of ENI (0 means the primary ENI)
 	DeviceNumber int
 	// IPv4Addresses shows whether each address is assigned, the key is IP address, which must
 	// be in dot-decimal notation with no leading zeros and no whitespace(eg: "10.1.0.253")
-	IPv4Addresses map[string]*AddressInfo
-	// Key is the prefix and value is prefix, used IPs under the prefix
-	IPv4Prefixes map[string]*ENIPrefix
+	// Key is the IP address - PD: "IP/28" and SIP: "IP/32"
+	AvailableIPv4Cidrs map[string]*AssignedIPv4Addresses
 }
 
 // AddressInfo contains information about an IP, Exported fields will be marshaled for introspection.
 type AddressInfo struct {
 	IPAMKey        IPAMKey
 	Address        string
 	UnassignedTime time.Time
-	Prefix         net.IP
 }
 
-func (e *ENI) findAddressForSandbox(ipamKey IPAMKey) *AddressInfo {
-	for _, addr := range e.IPv4Addresses {
-		if addr.IPAMKey == ipamKey {
-			return addr
+type AssignedIPv4Addresses struct {
```

Comment:
what "Assigned" means here, Assigned to ENI or Assigned to Pod?
seems here it's means "Assigned" to ENI, but in the `AssignedIPv4Addresses` function, it means assigned to pod.

maybe pick a better name 😄

Reply:
Yeah, I can rename maybe to AllocatedIPv4Addresses?

Reply:
I wondered this too, so I'm glad you raised it :)  They're not necessarily actually "assigned" or "allocated" :sweat_smile: and even in the `*ENI` struct we call them `AvailableIPv4Cidrs`

Maybe "AvailableIPv4Addresses" or "AttachedIPv4Addresses" - or skip the subjective term and just call it `CidrInfo` ?

The idea is that we have a 3-level nested structure that:
1. lists all the known ENIs (ENIPool - a collection of `*ENI`), that then includes a
2. list of all the current 'available' (or 'attached'?) known CIDRs (collection of ^this type), that tthen includes a
3. list of all the allocated (or in some version of cool-down) specific IPs.  This last one has to be sparse (we don't want to list _all_ the unallocated IPs) because an IPv6 CIDR is _huge_.

Reply:
During implementation I was thinking in terms of /32's assigned to pods within the Available Cidrs hence termed it as `AssignedIPv4Addresses` within `AvailableIPv4Cidrs`. I think we can make it `CidrInfo` and can reuse it for IPv6. 

Yes regarding [3], it is sparse and it includes just IPs assigned to pods and cool down IPs.

Reply:
Have updated the PR :)